packages/flutter/lib/src/scheduler/performance_mode.dart

```diff
@@ -0,0 +1,62 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+/// Used to request a specific [DartPerformanceMode].
+///
+/// When a request is made to the engine via the [PlatformDispatcher] it is
+/// not guaranteed to result in a specific [DartPerformanceMode]. This is best
+/// effort and the engine may choose to ignore the request.
+///
+/// This is currently not supported on web.
+class PerformanceModeHandler {
```

Comment:
I'd consider folding this into the SchedulerBinding instead of making a new class, as that will avoid introducing more semi-global variables and provides a mechanism for us to override/track this functionality in testing.